TESTS/icb_arpack_cpp.cpp

```diff
@@ -98,63 +93,64 @@ void real_symmetric_runner(double const& tol_check) {
       throw std::domain_error("Correct eigenvalues not computed");
     }
   }
-  std::cout << "------\n";
+  std::cout << "------" << std::endl;
 }
 
 template <typename Real>
 void diagonal_matrix_vector_product(std::complex<Real> const* const x,
                                     std::complex<Real>* const y) {
   for (int i = 0; i < 1000; ++i) {
-    y[i] = x[i] * std::complex<Real>{Real(i + 1), Real(i + 1)};
+    // Use complex matrix (i, -i) instead of (i, i): this way "largest_magnitude"
+    // and "largest_imaginary" options produce different results that can be checked.
+    y[i] = x[i] * std::complex<Real>{Real(i + 1), -Real(i + 1)};
   }
 }
 
 template <typename Real>
-void complex_symmetric_runner(double const& tol_check) {
+void complex_nonsymmetric_runner(double const& tol_check, arpack::which const& ritz_option) {
   a_int const N = 1000;
   a_int const nev = 9;
 
   a_int const ncv = 2 * nev + 1;
   a_int const ldv = N;
 
-  a_int const ldz = N + 1;
+  a_int const ldz = N;
 
-  a_int const lworkl = 3 * (ncv * ncv) + 6 * ncv;
+  a_int const lworkl = ncv * (3 * ncv + 5);
 
   Real const tol = 0.000001; // small tol => more stable checks after EV computation.
-  std::complex<Real> const sigma(0.0, 0.0);
+  std::complex<Real> const sigma(0.0, 0.0);  // not referenced in this mode
 
-  a_int const rvec = 0;
+  a_int const rvec = 0;      // eigenvectors omitted
 
   std::vector<std::complex<Real>> resid(N);
   std::vector<std::complex<Real>> V(ncv * N);
   std::vector<std::complex<Real>> workd(3 * N);
   std::vector<std::complex<Real>> workl(lworkl);
-  std::vector<std::complex<Real>> d(nev + 1);
-  std::vector<std::complex<Real>> z((N + 1) * (nev + 1));
+  std::vector<std::complex<Real>> d(nev);
+  std::vector<std::complex<Real>> z(ldz * nev);
   std::vector<Real> rwork(ncv);
   std::vector<std::complex<Real>> workev(2 * ncv);
+  std::vector<a_int> select(ncv); // since HOWMNY = 'A', only used as workspace here
 
-  std::array<a_int, 11> iparam{};
-  iparam[0] = 1;
-  iparam[2] = 10 * N;
-  iparam[3] = 1;
-  iparam[4] = 0;  // number of ev found by arpack.
```

Comment:
Can you let iparam[4] initialized to 0?
Initialisation is always good thing even if it's an output.

Reply:
For the initialization discussion: I am neutral in choosing to initialize (or not) the output-only arguments, but inclined to have a consistent treatment of all these arguments in the same function. If we choose to initialize iparam[4] because "initialization is always a good thing", by the same token we should also initialize all other elements in "iparam[0:10]", and also all elements in "ipntr[:]", and also all elements in the eigenvalues -- the list simply goes on... So I feel like just not necessary to do so just particularly for iparam[4], since it is not special at all compared to other variables mentioned above.

What do you think? If you want, I can still switch on ipram[4] initialization...